pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -314,6 +314,18 @@ func (c *Cloud) ensureLoadBalancer(namespacedName types.NamespacedName, loadBala
 				}
 			}
 		}
+
+		{
+			// Add additional tags
+			glog.V(2).Infof("Creating additional load balancer tags for %s", loadBalancerName)
```

Comment:
It would be nice to check that the tags aren't already present, both to avoid using up all our API operations quota, and to avoid this log message confusing users.  Perhaps in a follow up PR?

Reply:
(And that might allow harmonization with the existing create-map-of-tags code...)

Reply:
Yes, I did actually go down that route initially, adding a DescribeTags method and using the results of that to work out which tags were missing. But I stripped it out as I realised I was doing at best one API call each time (if nothing missing) and possibly two API calls (if something needed to be added). Whereas just adding them all each time is a single call, no matter what. Unless I've misunderstood something? 

I'd be happy to add the checking logic back in another PR though (would also allow, with a bit more work to make sure default tags were unaffected, for removing tags that are removed from the annotation).

Reply:
Ah - you're right.  I was thinking tags were returned by DescribeLoadBalancer.

Rate limits for read operations are typically much higher than for write operations, but I think we can wait and see if it's a problem in practice - particularly as when there are no additional tags we don't hit this case.

Reply:
Cool, I have all the code still, so if it's useful to add in, let me know.